generic.go

```diff
@@ -130,3 +191,52 @@ func hash(u uint64) uint64 {
 	u ^= u >> 33
 	return u
 }
+
+// GetHasher returns a default hasher function for different Key types.
+func GetHasher[Key any]() HashFn[Key] {
```

Comment:
It would be great to implement this without Unsafe, is that possible? Or better, remove the need for this by using `EqualsFn` and `HashFn` like the existing map.

Reply:
I removed the function and the unsafe import